skimage/morphology/gray.py

```diff
@@ -176,19 +115,41 @@ def erosion(image, footprint=None, out=None, shift_x=False, shift_y=False):
         The array to store the result of the morphology. If None is
         passed, a new array will be allocated.
     shift_x, shift_y : bool, optional
-        shift footprint about center point. This only affects
-        eccentric footprints (i.e. footprint with even numbered
-        sides).
+        Shift footprint about center point. This only affects 2D footprints
+        with even-numbered sides. These two parameters are deprecated in
+        favor of ``mirror``.
+
+        .. versionchanged:: 0.20
+            Parameters ``shift_x`` and ``shift_y`` are deprecated and will be
+            removed in a future version.
+
+    mirror : bool, optional
+        Mirror the footprint along each dimension. Default is ``False``.
+
+        .. versionadded:: 0.20
+            ``mirror`` was added in 0.20.
+
+    mode : {None, 'reflect','constant','nearest','mirror', 'wrap'}, optional
+        The ``mode`` parameter determines how the array borders are handled.
+        If ``None``, pixels outside the image domain are assumed to be the
+        maximum for the image's dtype, which causes them to not influence the
+        result. Default is ``'reflect'``.
+    cval : scalar, optional
+        Value to fill past edges of input if ``mode`` is 'constant'. Default
+        is 0.0.
+
+        .. versionadded:: 0.20
+            ``mode`` and ``cval`` were added in 0.20.
 
     Returns
     -------
-    eroded : array, same shape as `image`
+    eroded : array, same shape as ``image``
         The result of the morphological erosion.
 
     Notes
     -----
     For ``uint8`` (and ``uint16`` up to a certain bit-depth) data, the
-    lower algorithm complexity makes the `skimage.filters.rank.minimum`
+    lower algorithm complexity makes the ``skimage.filters.rank.minimum``
```

Comment:
```suggestion
    lower algorithm complexity makes the :func:`skimage.filters.rank.minimum`
```
Good opportunity to create a linked reference here (and elsewhere through this PR) if we are already changing it up.